### PR TITLE
feat: misc outbound edi cleanup

### PR DESCRIPTION
**Pull request by rsioss**
Changes are broken out into several commits, many of which just address some nit type of fixes. The 🍖  of the changes are in ef229004366fc0811faf68904713d90ef0a6fb46:

- use zod schema for event parsing
- move mapping invocation to shared lib
- refactor to support both array and object guide json
- remove hardcoded st control number
- add st control number validation

### CODE REVIEW

File: src/functions/edi/outbound/handler.ts [hunk 1]
```diff
@@ -122,46 +111,29 @@ export const handler = async (
       },
     };
 
-    const deliveryResults = [];
+    const deliveryResults = await Promise.all(transactionSetConfig.destinations.map(
```
Comment (BDQ) [inline src/functions/edi/outbound/handler.ts:114]: Should we use `allSettled` here instead? with `all` a single failure will halt the waiting for other promises.

Comment (rsioss) [reply src/functions/edi/outbound/handler.ts:114]: My initial thought is to stick with `all` here, which is consistent with the previous implementation (any rejections within the `for` would have stopped the processing as well). 

Happy to continue the discussion and do a follow-up if needed though.

Comment (BDQ) [reply src/functions/edi/outbound/handler.ts:114]: My point was more about knowing the state of the destinations promises, say you have 3x configured - one raises an exception - `.all` will resolve immediately and the state of the other two promises would be unknown.

**Comment by rsioss** [comment]
> Looks good, left one question on the code.
> 
> Also, where are control numbers coming from now?

Thanks, @BDQ! Replied to the code question in-line. Transaction set control numbers come from the input now -- I added the field to the guide JSON we have in the 855 `outbound.json`, and if folks are using a mapping it's expected that the mapping output includes this as well (since it's in the target schema).

See this [related discussion on Slack](https://stedi-inc.slack.com/archives/C02K6MA97L5/p1674493247400199), but the TL;DR is that the ST control number in the guide schema can be a string (older guides) or a number (newer guides), so hardcoding as a string is definitely problematic.

image

File: src/functions/edi/outbound/handler.ts (hunk 1 above)
Comment (rsioss) [reply src/functions/edi/outbound/handler.ts:114]: Is this different than the previous implementation with the `for` loop? I suppose it is slightly different in that the other destinations would not have even started if the first one fails, and now there is a chance that one is mid-flight when a failure happens. 

The current flow feels a bit odd as-is, due to the data model. We have one input, and we generate a single envelope, but are then potentially calling map/translate N times for each destination, and then doing the delivery to those destinations. IMO this flow seems better suited for inbound (where you might want to send inbound data to multiple APIs). For outbound, it feels like we should generate a single EDI document with the envelope that has been generated, and then potentially deliver that generated document to multiple destinations (perhaps an SFTP bucket and an archive bucket, for example).

Happy to do a follow up like I said, but I think it should possibly involve a bit more than just the change to `allSettled`. Would love to hear your thoughts!

**Pull request merged**